tests/lib/cli-engine/config-array/override-tester.js

```diff
@@ -0,0 +1,254 @@
+                it(`emits an error given '${pattern}'`, () => {
+                    let errorMessage;
+
+                    try {
+                        test(filePath, pattern);
+                    } catch (e) {
+                        errorMessage = e.message;
+                    }
+
+                    assert.strictEqual(errorMessage, expectedMessage);
+                });
+            }
+
+            // files in the project root
+            match("foo.js", ["foo.js"], []);
+            match("foo.js", ["*"], []);
+            match("foo.js", ["*.js"], []);
+            match("foo.js", ["**/*.js"], []);
+            match("bar.js", ["*.js"], ["foo.js"]);
+
+            noMatch("foo.js", ["./foo.js"], []);
+            noMatch("foo.js", ["./*"], []);
+            noMatch("foo.js", ["./**"], []);
+            noMatch("foo.js", ["*"], ["foo.js"]);
+            noMatch("foo.js", ["*.js"], ["foo.js"]);
+            noMatch("foo.js", ["**/*.js"], ["foo.js"]);
+
+            // files in a subdirectory
+            match("subdir/foo.js", ["foo.js"], []);
+            match("subdir/foo.js", ["*"], []);
+            match("subdir/foo.js", ["*.js"], []);
+            match("subdir/foo.js", ["**/*.js"], []);
+            match("subdir/foo.js", ["subdir/*.js"], []);
+            match("subdir/foo.js", ["subdir/foo.js"], []);
+            match("subdir/foo.js", ["subdir/*"], []);
+            match("subdir/second/foo.js", ["subdir/**"], []);
+
+            noMatch("subdir/foo.js", ["./foo.js"], []);
+            noMatch("subdir/foo.js", ["./**"], []);
+            noMatch("subdir/foo.js", ["./subdir/**"], []);
+            noMatch("subdir/foo.js", ["./subdir/*"], []);
+            noMatch("subdir/foo.js", ["*"], ["subdir/**"]);
+            noMatch("subdir/very/deep/foo.js", ["*.js"], ["subdir/**"]);
+            noMatch("subdir/second/foo.js", ["subdir/*"], []);
+            noMatch("subdir/second/foo.js", ["subdir/**"], ["subdir/second/*"]);
+
+            // error conditions
+            error("foo.js", ["/*.js"], "Invalid override pattern (expected relative path not containing '..'): /*.js");
+            error("foo.js", ["/foo.js"], "Invalid override pattern (expected relative path not containing '..'): /foo.js");
+            error("foo.js", ["../**"], "Invalid override pattern (expected relative path not containing '..'): ../**");
+        });
+    });
+
+    describe("'JSON.stringify(...)' should return readable JSON; not include 'Minimatch' objects", () => {
+        it("should return an object that has three properties 'includes', 'excludes', and 'basePath' if that 'patterns' property include one object.", () => {
+            const files = "*.js";
+            const excludedFiles = "test/*";
+            const basePath = process.cwd();
+            const tester = OverrideTester.create(files, excludedFiles, basePath);
+
+            assert.strictEqual(
+                JSON.stringify(tester),
```

Comment:
I think it would be better to test `.toJSON()` and check for presence or absence of properties in the object, but I don't feel strongly.